src/copilot/inspect/InspectionCopilot.ts

```diff
@@ -0,0 +1,256 @@
+                result = "HR";
+            } else {
+                result = "FTE";
+            }
+            return result;
+        }
+        public void test(String[] arr) {
+            try {
+                Integer.parseInt(arr[0]);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    `;
+    public static readonly EXAMPLE_ASSISTANT_MESSAGE = `
+    @Entity
+    // @PROBLEM: Using a traditional POJO
+    // @SOLUTION: transform into a record
+    // @SYMBOL: EmployeePojo
+    // @SEVERITY: MIDDLE
+    public class EmployeePojo implements Employee {
+        public final String name;
+        public EmployeePojo(String name) {
+            this.name = name;
+        }
+        public String getRole() {
+            String result = '';
+            // @PROBLEM: Using if-else statements to check the type of animal
+            // @SOLUTION: Use switch expression
+            // @SYMBOL: if
+            // @SEVERITY: MIDDLE
+            if (this.name.equals("Miller")) {
+                result = "Senior";
+            } else if (this.name.equals("Mike")) {
+                result = "HR";
+            } else {
+                result = "FTE";
+            }
+            return result;
+        }
+        public void test(String[] arr) {
+            try {
+                Integer.parseInt(arr[0]);
+            } catch (Exception e) {
+                // @PROBLEM: Print stack trace in case of an exception
+                // @SOLUTION: Log errors to a logger
+                // @SYMBOL: ex.printStackTrace
+                // @SEVERITY: LOW
+                e.printStackTrace();
+            }
+        }
+    }
+    //${Copilot.DEFAULT_END_MARK}
+    `;
+
+    // Initialize regex patterns
+    private static readonly PROBLEM_PATTERN: RegExp = /\/\/ @PROBLEM: (.*)/;
+    private static readonly SOLUTION_PATTERN: RegExp = /\/\/ @SOLUTION: (.*)/;
+    private static readonly SYMBOL_PATTERN: RegExp = /\/\/ @SYMBOL: (.*)/;
+    private static readonly LEVEL_PATTERN: RegExp = /\/\/ @SEVERITY: (.*)/;
+
+    private readonly debounceMap = new Map<string, NodeJS.Timeout>();
+
+    public constructor() {
+        const messages: { role: string, content: string }[] = [
+            { role: "system", content: InspectionCopilot.SYSTEM_MESSAGE },
+            { role: "user", content: InspectionCopilot.EXAMPLE_USER_MESSAGE },
+            { role: "assistant", content: InspectionCopilot.EXAMPLE_ASSISTANT_MESSAGE },
+        ];
+        super(messages);
+    }
+
+    /**
+     * inspect the given code using copilot and return the inspections
+     * @param code code to inspect
+     * @param key key to debounce the inspecting, if provided
+     * @param wait debounce time in milliseconds, default is 3000ms
+     * @returns inspections provided by copilot
+     */
+    public inspectCode(code: string, key?: string, wait?: number): Promise<Inspection[]> {
```

Comment:
don't understand what does the `key` mean?

Reply:
consider the case that we have multiple documents(A, B, C), and we only want to debounce the method calls on the same `key` (document).